src/main/java/io/cloudsoft/terraform/TerraformConfigurationImpl.java

```diff
@@ -74,7 +72,22 @@ protected void preStop() {
     @Override
     protected void postStop() {
         getChildren().forEach(c -> c.sensors().set(Attributes.SERVICE_STATE_ACTUAL, Lifecycle.STOPPED));
-        getChildren().forEach(child -> removeChild(child));
+        getChildren().forEach(child -> {
+            if (child instanceof BasicGroup){
+                child.getChildren().forEach(grandChild -> {
+                    if (grandChild instanceof ManagedResource || grandChild instanceof DataResource){
+                        removeChild(grandChild);
+                        Entities.unmanage(grandChild);
```

Comment:
`unmanage` takes care of removing entity, no?

Reply:
yes, and does it recursively too...

Reply:
turns out it doesn't, so the remove/unmanage call pair is what is needed